File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupResult.java
package closer.vlllage.com.closer.api.models;

import com.google.gson.annotations.SerializedName;

import java.util.List;

import closer.vlllage.com.closer.store.models.Group;

import static java.lang.Boolean.TRUE;

public class GroupResult extends ModelResult {
    public String name;
    public String about;
    @SerializedName("public") public Boolean isPublic;
    public Boolean physical;
    public Boolean hub;
    public String eventId;
    public List<Double> geo;

    public static Group from(GroupResult groupResult) {
        Group group = new Group();
        group.setId(groupResult.id);
        updateFrom(group, groupResult);
        return group;
    }

    public static Group updateFrom(Group group, GroupResult groupResult) {
        group.setName(groupResult.name);
        group.setUpdated(groupResult.updated);
        group.setAbout(groupResult.about);
        group.setPublic(TRUE.equals(groupResult.isPublic));
        group.setPhysical(TRUE.equals(groupResult.physical));
        group.setHub(TRUE.equals(groupResult.hub));
        group.setEventId(groupResult.eventId);

        if (groupResult.geo != null && !groupResult.geo.isEmpty()) {
            group.setLatitude(groupResult.geo.get(0));
            group.setLongitude(groupResult.geo.get(1));
        }

        return group;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/GroupResult.kt
package closer.vlllage.com.closer.api.models

import closer.vlllage.com.closer.store.models.Group
import com.google.gson.annotations.SerializedName

class GroupResult : ModelResult() {
    var name: String? = null
    var about: String? = null
    @SerializedName("public")
    var isPublic: Boolean? = null
    var physical: Boolean? = null
    var hub: Boolean? = null
    var eventId: String? = null
    var phoneId: String? = null
    var groupMessageId: String? = null
    var ratingAverage: Double? = null
    var ratingCount: Int? = null
    var geo: List<Double>? = null
    private val photo: String? = null

    companion object {

        fun from(groupResult: GroupResult): Group {
            val group = Group()
            group.id = groupResult.id
            updateFrom(group, groupResult)
            return group
        }

        fun updateFrom(group: Group, groupResult: GroupResult): Group {
            group.name = groupResult.name
            group.updated = groupResult.updated
            group.about = groupResult.about
            group.ratingAverage = groupResult.ratingAverage
            group.ratingCount = groupResult.ratingCount
            group.isPublic = true == groupResult.isPublic
            group.physical = true == groupResult.physical
            group.hub = true == groupResult.hub
            group.eventId = groupResult.eventId
            group.phoneId = groupResult.phoneId
            group.groupMessageId = groupResult.groupMessageId
            group.photo = groupResult.photo

            if (groupResult.geo.isNullOrEmpty().not()) {
                group.latitude = groupResult.geo!![0]
                group.longitude = groupResult.geo!![1]
            }

            return group
        }
    }
}
